lib/node_modules/@stdlib/array/fixed-endian-factory/test/test.to_reversed.js

```diff
@@ -0,0 +1,120 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var hasOwnProp = require( '@stdlib/assert/has-own-property' );
+var isFunction = require( '@stdlib/assert/is-function' );
+var instanceOf = require( '@stdlib/assert/instance-of' );
+var Uint8Array = require( '@stdlib/array/uint8' );
+var factory = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof factory, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the main export is a `toReversed` method', function test( t ) {
+	var ctor = factory( 'float64' );
+	t.strictEqual( hasOwnProp( ctor.prototype, 'toReversed' ), true, 'returns expected value' );
+	t.strictEqual( isFunction( ctor.prototype.toReversed ), true, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a typed array instance', function test( t ) {
+	var values;
+	var ctor;
+	var arr;
+	var i;
+
+	ctor = factory( 'float64' );
+	arr = new ctor( 'little-endian', 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.toReversed.call( value );
+		};
+	}
+	t.end();
+});
+
+tape( 'the method returns an empty array if operating on an empty typed array', function test( t ) {
+	var ctor;
+	var arr;
+	var out;
+
+	ctor = factory( 'float64' );
+	arr = new ctor( 'little-endian' );
+	out = arr.toReversed();
+
+	t.strictEqual(out.length, 0, 'returns expected value');
+	t.end();
+});
+
+tape( 'the method returns a new typed array containing elements in reverse order', function test( t ) {
+	var expected;
+	var ctor;
+	var arr;
+	var out;
+
+	ctor = factory( 'float64' );
+	arr = new ctor( 'little-endian', [ 1.0, 2.0, 3.0, 4.0 ] );
+	expected = new Uint8Array( [ 4.0, 3.0, 2.0, 1.0 ] );
+	out = arr.toReversed();
+
+	t.strictEqual( instanceOf( out, ctor ), true, 'returns expected value' );
+	t.notEqual( out, arr, 'returns a new instance' );
+	t.strictEqual( out.length, expected.length, 'returns expected value' );
```

Comment:
this test doesn't compare values of `expected` against values of `out`, it just compares the length.

Reply:
made changes